### ISSUE TITLE
In centos7.3, the folder is parsed into a file name when it is uncompressed.

### ISSUE THREAD

**Issue by langjxk** [srikanth-lingala/zip4j#81]
In centos7.3, the folder is parsed into a file name when it is uncompressed. For example: "home/myzip/ziptest.zip" should be extracted to "home/myzip/ziptest/test/mytest1.zip", but it is incorrectly Unzip it to "home/myzip/ziptest\test\mytest1.zip", where "ziptest\test\mytest1.zip" becomes the file name

**Comment by srikanth-lingala** [comment]
I just made a quick fix and made a SNAPSHOT release (fix is not yet on github). 
Can you please check with version `2.2.3-SNAPSHOT` and see if it works fine for you? If yes, I will make push the fix and make a release.

**Comment by langjxk** [comment]
> I just made a quick fix and made a SNAPSHOT release (fix is not yet on github).
> Can you please check with version `2.2.3-SNAPSHOT` and see if it works fine for you? If yes, I will make push the fix and make a release.

How do I get 2.2.3-SNAPSHOT? I can't get it from maven.

**Comment by srikanth-lingala** [comment]
Please use this in your pom. It is also totally fine if you just use the `snapshotRepository` part of it:

```
<distributionManagement>
        <snapshotRepository>
            <id>ossrh</id>
            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
        </snapshotRepository>
        <repository>
            <id>ossrh</id>
            <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
        </repository>
    </distributionManagement>
```

**Comment by langjxk** [comment]
The bug is still not resolved. In version 1.3.2, it can be solved as follows:
`if(File.separator.equals("/")){
				List<FileHeader> headerListModify = zFile.getFileHeaders();
				for (FileHeader fileHeader : headerListModify) {
					if (StringUtils.isNotBlank(fileHeader.getFileName())) {
						fileHeader.setFileName(fileHeader.getFileName().replaceAll("\\\\",File.separator));
					}
				}
			}`
However, the following verification has been added to the 2.x version, which makes it impossible to continue using this method.
`private void verifyNextEntry(ZipInputStream zipInputStream, FileHeader fileHeader) throws IOException {
    LocalFileHeader localFileHeader = zipInputStream.getNextEntry(fileHeader);

    if (localFileHeader == null) {
      throw new ZipException("Could not read corresponding local file header for file header: "
          + fileHeader.getFileName());
    }

    if (!fileHeader.getFileName().equals(localFileHeader.getFileName())) {
      throw new ZipException("File header and local file header mismatch");
    }
  }`

**Comment by langjxk** [comment]
The bug is still not resolved. In version 1.3.2, it can be solved as follows:

if(File.separator.equals("/")){ 
    List<FileHeader> headerListModify = zFile.getFileHeaders();
    for (FileHeader fileHeader : headerListModify) { 
        if (StringUtils.isNotBlank(fileHeader.getFileName())) { 
            fileHeader.setFileName(fileHeader.getFileName().replaceAll("\\\\",File.separator)); 
        } 
    } 
}
 
However, the following verification has been added to the 2.x version, which makes it impossible to continue using this method

private void verifyNextEntry(ZipInputStream zipInputStream, FileHeader fileHeader) throws IOException {
    LocalFileHeader localFileHeader = zipInputStream.getNextEntry(fileHeader);
	
    if (localFileHeader == null) {
        throw new ZipException("Could not read corresponding local file header for file header: "+ fileHeader.getFileName());
    }

    if (!fileHeader.getFileName().equals(localFileHeader.getFileName())) {
        throw new ZipException("File header and local file header mismatch");
    }
}

**Comment by srikanth-lingala** [comment]
With this snapshot version, do you get any error message or do you still get the output file names as `home/myzip/ziptest\test\mytest1.zip`?

**Comment by langjxk** [comment]
The output file name is still "home/myzip/ziptest\test\mytest1.zip".

**Comment by srikanth-lingala** [comment]
Can you try with `2.2.3.1-SNAPSHOT`?

**Comment by langjxk** [comment]
2.2.3.1-SNAPSHOT still has bugs. In centos7.3, the folder will not be created when decompressing. Once the multi-level directory is extracted, it will report an error: "ZipException!java.io.FileNotFoundException". For example: "net.lingala.zip4j.exception.ZipException: java.io.FileNotFoundException: home/myzip/ziptest/dir1/test.txt
At net.lingala.zip4j.tasks.AsyncZipTask.performTaskWithErrorHandling(AsyncZipTask.java:49)
At net.lingala.zip4j.tasks.AsyncZipTask.execute(AsyncZipTask.java:36)
At net.lingala.zip4j.ZipFile.extractAll(ZipFile.java:437)"
In windows, another error will be reported: "net.lingala.zip4j.exception.ZipException: java.lang.IllegalArgumentException: character to be escaped is missing
At net.lingala.zip4j.tasks.AsyncZipTask.performTaskWithErrorHandling(AsyncZipTask.java:49)
At net.lingala.zip4j.tasks.AsyncZipTask.execute(AsyncZipTask.java:36)
At net.lingala.zip4j.ZipFile.extractAll(ZipFile.java:437)"

**Comment by srikanth-lingala** [comment]
I made some more changes. Can you please try again with `2.2.3.3-SNAPSHOT`?

**Comment by langjxk** [comment]
2.2.3.3-SNAPSHOT is already available in centos7.3, but it is still the last bug in win10: "net.lingala.zip4j.exception.ZipException: java.lang.IllegalArgumentException: character to be escaped is missing
At net.lingala.zip4j.tasks.AsyncZipTask.performTaskWithErrorHandling(AsyncZipTask.java:49)
At net.lingala.zip4j.tasks.AsyncZipTask.execute(AsyncZipTask.java:36)
At net.lingala.zip4j.ZipFile.extractAll(ZipFile.java:437)"

**Comment by srikanth-lingala** [comment]
Can you please try again with `2.2.3.4-SNAPSHOT`?

**Comment by langjxk** [comment]
After testing, 2.2.3.4-SNAPSHOT found no bugs in centos7.3 and win10. thanks.

**Comment by srikanth-lingala** [comment]
Fixed in v2.2.3
